#! /bin/sh /usr/share/dpatch/dpatch-run
## 28_irrecord_resume_support.dpatch by Mathias Hasselmann
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Adds resume support to irrecord (LP: #197493)

@DPATCH@
diff -urNad lirc-0.8.3~pre1~/daemons/irrecord.c lirc-0.8.3~pre1/daemons/irrecord.c
--- lirc-0.8.3~pre1~/daemons/irrecord.c	2008-03-24 00:11:51.000000000 -0500
+++ lirc-0.8.3~pre1/daemons/irrecord.c	2008-03-24 00:11:59.000000000 -0500
@@ -192,6 +192,7 @@
 	lirc_t min_remaining_gap, max_remaining_gap;
 	int force;
 	int retries;
+	int resume;
 	struct ir_remote *remotes=NULL;
 	char *device=NULL;
 #ifdef DEBUG
@@ -200,6 +201,7 @@
 
 	progname=argv[0];
 	force=0;
+	resume=0;
 	hw_choose_driver(NULL);
 	while(1)
 	{
@@ -211,6 +213,7 @@
 			{"device",required_argument,NULL,'d'},
 			{"driver",required_argument,NULL,'H'},
 			{"force",no_argument,NULL,'f'},
+			{"resume",no_argument,NULL,'r'},
 #ifdef DEBUG
 			{"pre",no_argument,NULL,'p'},
 			{"post",no_argument,NULL,'P'},
@@ -221,9 +224,9 @@
 			{0, 0, 0, 0}
 		};
 #ifdef DEBUG
-		c = getopt_long(argc,argv,"hvd:H:fpPtiT",long_options,NULL);
+		c = getopt_long(argc,argv,"hvd:H:frpPtiT",long_options,NULL);
 #else
-		c = getopt_long(argc,argv,"hvd:H:f",long_options,NULL);
+		c = getopt_long(argc,argv,"hvd:H:fr",long_options,NULL);
 #endif
 		if(c==-1)
 			break;
@@ -234,6 +237,7 @@
 			printf("\t -h --help\t\tdisplay this message\n");
 			printf("\t -v --version\t\tdisplay version\n");
 			printf("\t -f --force\t\tforce raw mode\n");
+			printf("\t -r --resume\t\tcontinue recording\n");
 			printf("\t -H --driver=driver\tuse given driver\n");
 			printf("\t -d --device=device\tread from given device\n");
 			exit(EXIT_SUCCESS);
@@ -254,6 +258,9 @@
 		case 'f':
 			force=1;
 			break;
+		case 'r':
+			resume=1;
+			break;
 #ifdef DEBUG
 		case 'p':
 			get_pre=1;
@@ -459,7 +466,7 @@
 	switch(hw.rec_mode)
 	{
 	case LIRC_MODE_MODE2:
-		if(remotes==NULL && !get_lengths(&remote,force))
+		if((!remotes || !resume) && !get_lengths(&remote,force))
 		{
 			if(remote.gap==0)
 			{
@@ -493,7 +500,7 @@
 	case LIRC_MODE_LIRCCODE:
 		if(hw.rec_mode==LIRC_MODE_CODE) remote.bits=CHAR_BIT;
 		else remote.bits=hw.code_length;
-		if(!get_gap_length(&remote))
+		if((!remotes || !resume) && !get_gap_length(&remote))
 		{
 			fprintf(stderr,"%s: gap not found,"
 				" can't continue\n",progname);
@@ -766,7 +773,7 @@
 		exit(EXIT_FAILURE);
 	}
 	
-	if(!has_toggle_bit_mask(remotes))
+	if((!remotes || !resume) && !has_toggle_bit_mask(remotes))
 	{
 		get_toggle_bit_mask(remotes);
 	}
